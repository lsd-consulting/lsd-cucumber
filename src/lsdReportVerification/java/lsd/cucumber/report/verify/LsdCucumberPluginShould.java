package lsd.cucumber.report.verify;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static java.nio.file.Paths.get;
import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;

class LsdCucumberPluginShould {

    @Test
    void generateMenu() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/lsd-index.html"));
    }

    @Test
    void generateReportForBasicScenario() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario.html"));
    }

    @Test
    void generateReportForScenarioWithDataTable() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Data_table.html"));
    }

    @Test
    void generateForScenarioOutline() throws IOException {
        Approvals.verify(copyOfFile("build/reports/lsd/Scenario_outline.html"));
    }

    private File copyOfFile(String path) throws IOException {
        Path original = get(path);
        Path copy = Path.of(original.getParent().toString(), "copy_" + original.getFileName());
        Files.copy(original, copy, REPLACE_EXISTING);
        return copy.toFile();
    }
}
