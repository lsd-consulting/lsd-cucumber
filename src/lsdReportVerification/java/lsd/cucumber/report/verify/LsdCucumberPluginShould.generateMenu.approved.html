<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/html" xml:lang="en" lang="en">
<head>
    <title>Index</title>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/lsd-consulting/lsd-core@7/src/main/resources/static/style.min.css">

</head>
<body>
<main role="main">
    <h1 class="logo">LSD</h1>
    <h2><label class="index">Available Reports</label></h2>
    <div class="index-content">
        <a class="success" href="Data_table.html">data_table</a>
        <a class="success" href="Scenario.html">scenario</a>
        <a class="success" href="Scenario_outline.html">scenario_outline</a>
        
    </div>
</main>
</body>
</html>